api/app.go

```diff
@@ -306,17 +313,53 @@ func (app *App) AddError() {
 	app.Errors.Update(1)
 }
 
+func buildAddress(port int) string {
+	if port == 0 {
+		return "localhost:"
+	} else {
+		return fmt.Sprintf("localhost:%d", port)
+	}
+}
+
 // Start starts listening for web requests at specified host and port
 func (app *App) Start() error {
 	l := app.Logger.With(
 		zap.String("source", "app"),
 		zap.String("operation", "Start"),
 		zap.String("host", app.Host),
-		zap.Int("port", app.Port),
+		zap.Int("port", app.HTTPPort),
 	)
 
+	grpcLis, err := net.Listen("tcp", buildAddress(app.GRPCPort))
+	if err != nil {
+		return fmt.Errorf("error trying to listen for connections: %v", err)
+	}
+	app.GRPCEndpoint = grpcLis.Addr().String()
```

Comment:
Suggestion: I'd much rather have grcp_endpoint and http_endpoint be their own flags and drop the Host/HTTPPort/GRPCPort altogether. It kills a bunch of logic we don't need and makes each endpoint its own flag, instead of being split into multiple parts.

Up to you and @cscatolini , but I had to say it. :)

Reply:
I gave an additional though about this and I decided to go for the less logic style of only using HTTP and GRPC Endpoints.